### PR TITLE
📝 Documentation for single source of truth and diagram for data-flow

### PR DESCRIPTION
**Pull request by madchutney**
### Description

- Add documentation describing the single source of truth concept and add a diagram for data-flow used in mbed-targets
- Render native PlantUML in markdown using proxy service

Note: Since the source diagram is not yet in master the rendered document can be found here: https://github.com/ARMmbed/mbed-targets/blob/4bdd75147d2fed1a71071ca80c0e04a577a5c5cc/DATA_VALIDATION.md

### Test Coverage

- [ ]  This change is covered by existing or additional automated tests.
- [ ]  Manual testing has been performed (and evidence provided) as automated testing was not feasible.
- [x]  Additional tests are not required for this change (e.g. documentation update).

### CODE REVIEW

File: DATA_VALIDATION.md [hunk 1]
```diff
@@ -1,5 +1,41 @@
...
+| Tools Platform Database | [mbed-os-tools](https://github.com/ARMmbed/mbed-os-tools/blob/master/src/mbed_os_tools/detect/platform_database.py) | `mbed-ls` `mbed detect` |
+| `targets.json` | [mbed-os](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json) | `mbed compile`, Online Compiler |
+| Online Database | OS Site Database (os.mbed.com) | Product Code reservation, Public Websites, Mbed Studio |
+ 
+## Single Source of Truth
+
+The tools for Mbed 6 use a single source of truth for each piece of data, although all data may not come from the same
+source. For all common data, the source will be the OS Site Online Database, for example the `Product Code` to 
+`Build Target` mapping will be stored in the OS Site Database and all other definitions will be considered irrelevant.
+Other information, such as build configuration,  will continue to be stored in Mbed OS as it is only relevant to Mbed
```
Comment (rwalton-arm) [inline DATA_VALIDATION.md]:
```suggestion
Other information, such as build configuration, will continue to be stored in Mbed OS as it is only relevant to Mbed 
```

File: DATA_VALIDATION.md [hunk 2]
```diff
@@ -1,5 +1,41 @@
...
\ No newline at end of file
+Mbed has historically had a number of ways to manage target information, which has led to inconsistencies over time
+where some sources have been updated while others have not. This repository contains a tool that compares the various
+sources of data and generates a report of their consistency. This report is generated on a daily basis and this is
+available via the link above.
+
+The following table describes the data sources and usage for Mbed 5.
+
+| Data Source | Definition Location | Uses |
+| :---------- | :------------------ | :--- |
```
Comment (muchzill4) [inline DATA_VALIDATION.md]:
```suggestion
| Data Source | Definition Location | Used By |
| :---------- | :------------------ | :------ |
```

File: news/627.misc [hunk 3]
```diff
@@ -0,0 +1 @@
+Add documentation describing the single source of truth concept and add a diagram for data-flow used in mbed-targets
```
Comment (muchzill4) [inline news/627.misc]: `create-news-file` util to the rescue. If you're using latest version of `mbed-tools-ci-scripts` it should be available in the venv. :)

Comment (madchutney) [reply news/627.misc]: Good to know, in this case I was just using the Jira number

**Pull request merged**